frontend/src/app/main/ui/auth/register.cljs

```diff
@@ -18,19 +18,68 @@
    [app.main.ui.auth.login :as login]
    [app.main.ui.components.forms :as fm]
    [app.main.ui.components.link :as lk]
-   [app.main.ui.icons :as i]
    [app.util.i18n :as i18n :refer [tr]]
    [app.util.storage :as storage]
    [beicon.v2.core :as rx]
    [rumext.v2 :as mf]))
 
 ;; --- PAGE: Register
 
+(mf/defc newsletter-options
+  {::mf/props :obj
+   ::mf/private true}
+  []
+  (let [updates-label
+        (mf/html
+         [:> i18n/tr-html*
+          {:tag-name "div"
+           :content (tr "onboarding-v2.newsletter.updates")}])
+        news-label
+        (mf/html
+         [:> i18n/tr-html*
+          {:tag-name "div"
+           :content (tr "onboarding-v2.newsletter.news")}])]
+    [:* [:div {:class (stl/css :fields-row :input-visible :newsletter-option-wrapper)}
+         [:& fm/input {:name :accept-newsletter-updates
+                       :type "checkbox"
+                       :default-checked false
+                       :label updates-label}]]
+
+     [:div {:class (stl/css :fields-row :input-visible :newsletter-option-wrapper)}
+      [:& fm/input {:name :accept-newsletter-news
+                    :type "checkbox"
+                    :default-checked false
+                    :label news-label}]]]))
+
+(mf/defc terms-and-privacy
+  {::mf/props :obj
+   ::mf/private true}
+  []
+  (let [terms-label
+        (mf/html
+         [:> i18n/tr-html*
+          {:tag-name "div"
+           :content (tr "auth.terms-and-privacy-agreement"
+                        cf/terms-of-service-uri
+                        cf/privacy-policy-uri)}])]
+
+    [:div {:class (stl/css :fields-row :input-visible :accept-terms-and-privacy-wrapper)}
+     [:& fm/input {:name :accept-terms-and-privacy
+                   :class (stl/css :checkbox-terms-and-privacy)
+                   :type "checkbox"
+                   :default-checked false
+                   :label terms-label}]]))
+
 (def ^:private schema:register-form
   [:map {:title "RegisterForm"}
    [:password ::sm/password]
+   [:fullname [::sm/text {:max 250}]]
    [:email ::sm/email]
-   [:invitation-token {:optional true} ::sm/text]])
+   [:accept-terms-and-privacy {:optional (not (contains? cf/flags :terms-and-privacy-checkbox))}
+    [:and :boolean [:= true]]]
+   [:accept-newsletter-updates {:optional true :default false} [:and :boolean [:= true]]]
```

Comment:
The `:default` has no meaning here on specs. 

Also, I think `[:= true]` is also not necessary, the user should be able to check it and uncheck it (setting false value) with no issue.